src/core/utils/color/ColorValue.ts

```diff
@@ -0,0 +1,709 @@
+ * ```ts
+ *  const semiTransparentBlue = ColorValue.blue.withTransparency(100);
+ * ```
+ * @public
+ */
+export class ColorValue {
+  private readonly _tbgr: number;
+
+  public constructor(tbgr: number) {
+    scratchUInt32[0] = tbgr; // Force to be a 32-bit unsigned integer
+    this._tbgr = scratchUInt32[0];
+  }
+
+  /**
+   * Create a new ColorValue.
+   * @param val value to use.
+   * If a number, it is interpreted as a 0xTTBBGGRR (Red in the low byte, high byte is transparency 0==fully opaque) value.
+   *
+   * If a string, must be in one of the following forms:
+   * *"rgb(255,0,0)"*
+   * *"rgba(255,0,0,.2)"*
+   * *"rgb(100%,0%,0%)"*
+   * *"hsl(120,50%,50%)"*
+   * *"#rrggbb"*
+   * *"blanchedAlmond"* (see possible values from [[ColorByName]]). Case insensitive.
+   */
+  public static create(val?: string | ColorDefProps) {
+    return this.fromTbgr(this.computeTbgr(val));
+  }
+
+  /** @internal */
+  public static computeTbgr(val?: string | ColorDefProps): number {
+    switch (typeof val) {
+      case 'number':
+        return val;
+      case 'string':
+        return this.computeTbgrFromString(val);
+      default:
+        return 0;
+    }
+  }
+
+  /** Convert this ColorValue to a 32 bit number representing the 0xTTBBGGRR value */
+  public toJSON(): ColorDefProps {
+    return this._tbgr;
+  }
+
+  /** Create a new ColorValue from a json object. If the json object is a number, it is assumed to be a 0xTTBBGGRR value. */
+  public static fromJSON(json?: ColorDefProps): ColorValue {
+    return this.create(json);
+  }
+
+  /** Create a ColorValue from Red, Green, Blue, Transparency values. All inputs should be integers between 0-255. */
+  public static from(
+    red: number,
+    green: number,
+    blue: number,
+    transparency?: number,
+  ): ColorValue {
+    return this.fromTbgr(
+      this.computeTbgrFromComponents(red, green, blue, transparency),
+    );
+  }
+
+  /** Compute the 0xTTBBGGRR value corresponding to the specified Red, Green, Blue, Transparency components. All inputs should be integers between 0-255. */
+  public static computeTbgrFromComponents(
+    red: number,
+    green: number,
+    blue: number,
+    transparency?: number,
+  ): number {
+    scratchBytes[0] = red;
+    scratchBytes[1] = green;
+    scratchBytes[2] = blue;
+    scratchBytes[3] = transparency || 0;
+    return scratchUInt32[0];
+  }
+
+  /** Create a ColorValue from its 0xTTBBGGRR representation. */
+  public static fromTbgr(tbgr: number): ColorValue {
+    return new ColorValue(tbgr);
+  }
+
+  /** Create a ColorValue from a string representation. The following representations are supported:
+   * *"rgb(255,0,0)"*
+   * *"rgba(255,0,0,.2)"*
+   * *"rgb(100%,0%,0%)"*
+   * *"hsl(120,50%,50%)"*
+   * *"#rrbbgg"*
+   * *"blanchedAlmond"* (see possible values from [[ColorByName]]). Case-insensitive.
+   */
+  public static fromString(val: string): ColorValue {
```

Comment:
Yeah, same here. We could just support all of that in constructor.

Reply:
see new ColorValue.create method.  constructor method is now private and compatible with class in imodeljs... its just as easy to use `ColorValue.create()` instead of `new ColorValue ()`